TwoNote/app/src/main/java/com/microsoft/device/display/samples/twonote/NoteListFragment.kt

```diff
@@ -122,41 +131,37 @@ class NoteListFragment : Fragment(), AdapterView.OnItemClickListener, AdapterVie
 
         arrayAdapter?.getItem(position)?.let { inode ->
             activity?.let { activity ->
-                var note = loadNote("", "/n" + inode.id)
+                var note = fileHandler.loadNote(requireContext(),"", "/n" + inode.id)
                 if (note == null)
                     note = Note(inode.id)
 
                 if (ScreenHelper.isDualMode(activity)) {
                     parentFragmentManager.beginTransaction()
                         .replace(
                             R.id.dual_screen_end_container_id,
-                            NoteDetailFragment.newInstance(note), null
+                            NoteDetailFragment.newInstance(inode, note), null
                         ).commit()
                 } else {
                     parentFragmentManager.beginTransaction()
                         .replace(
                             R.id.single_screen_container_id,
-                            NoteDetailFragment.newInstance(note), null
+                            NoteDetailFragment.newInstance(inode, note), null
                         ).addToBackStack(null)
                         .commit()
                 }
             }
         }
     }
 
-    fun updateINode(title: String) {
-        var inode: INode?
+    fun updateINode(inode: INode, title: String) {
+        val position = DataProvider.inodes.indexOf(inode)
```

Comment:
more of a comment/note for the future, we need to make sure that the inode and listview indices match!